From cb4cf770d993d14e87ee20a74f66bc4cec77b528 Mon Sep 17 00:00:00 2001
From: Philippe Simons <simons.philippe@gmail.com>
Date: Wed, 14 Aug 2024 19:01:59 +0200
Subject: [PATCH] rg35xx: enable gpu

---
 .../sun50i-h700-anbernic-rg35xx-2024.dts      | 36 +++++++++++++++++++
 1 file changed, 36 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts
index 99974e28c..37f852f4d 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts
@@ -224,12 +224,62 @@ panel_in_rgb: endpoint {
 			};
 		};
 	};
+
+//	gpu_opp_table: opp-table-1 {
+//		compatible = "operating-points-v2";
+//
+//		opp-420000000 {
+//			opp-hz = /bits/ 64 <420000000>;
+//			opp-microvolt = <940000>;
+//		};
+//		opp-456000000 {
+//			opp-hz = /bits/ 64 <456000000>;
+//			opp-microvolt = <940000>;
+//		};
+//		opp-504000000 {
+//			opp-hz = /bits/ 64 <504000000>;
+//			opp-microvolt = <940000>;
+//		};
+//		opp-552000000 {
+//			opp-hz = /bits/ 64 <552000000>;
+//			opp-microvolt = <940000>;
+//		};
+//		opp-600000000 {
+//			opp-hz = /bits/ 64 <600000000>;
+//			opp-microvolt = <940000>;
+//		};
+//		opp-648000000 {
+//			opp-hz = /bits/ 64 <648000000>;
+//			opp-microvolt = <940000>;
+//		};
+//	};
 };
 
 &cpu0 {
 	cpu-supply = <&reg_dcdc1>;
 };
 
+&gpu {
+	clocks = <&ccu /* CLK_PLL_GPU */ 8>, <&ccu CLK_GPU0>, <&ccu CLK_GPU1>, <&ccu CLK_BUS_GPU>;
+	clock-names = "clk_parent", "clk_mali", "clk_bak", "clk_bus";
+	gpu_idle = <1>;
+	dvfs_status = <1>;
+	operating-points = <0x9e340 0xea600 0x927c0 0xdbba0 0x86c40 0xdbba0 0x7b0c0 0xdbba0 0x6f540 0xdbba0 0x668a0 0xdbba0>;
+   #cooling-cells = <2>;
+	power_policy = "always_on";
+	status = "okay";
+
+	ipa_dvfs {
+		compatible = "arm,mali-simple-power-model";
+		static-coefficient = <0x4268>;
+		dynamic-coefficient = <0x2ee>;
+		ts = <0x3e2da 0x2568 0xffffff98 0x04>;
+		thermal-zone = "gpu-thermal";
+		ss-coefficient = <0x24>;
+		ff-coefficient = <0x123>;
+	};
+};
+
 &de {
 	status = "okay";
 };
-- 
2.34.1

